pkg/workqueue/workqueue.proto

```diff
@@ -26,6 +27,37 @@ message ProcessResponse {
   int64 requeue_after_seconds = 1;
 }
 
+message GetKeyStateRequest {
+  // The key to retrieve information for
+  string key = 1;
+}
+
+message KeyState {
+  // The key name
+  string key = 1;
+
+  // Current status of the key
+  enum Status {
+    UNKNOWN = 0;
+    QUEUED = 1;
+    IN_PROGRESS = 2;
+    DEAD_LETTER = 3;
+  }
+  Status status = 2;
```

Comment:
We should have clear semantics for when a key is both queued and in-progress, ideally covered by conformance tests.

Reply:
Added!